IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -443,44 +567,135 @@ - (MSIDCredentialCacheItem *)tokenWithKey:(__unused MSIDCacheKey *)key
 
 // Read one or more credentials from the keychain that match the key (see credentialItem:matchesKey).
 // If not found, return an empty list without setting an error.
-- (NSArray<MSIDCredentialCacheItem *> *)tokensWithKey:(__unused MSIDCacheKey *)key
-                                           serializer:(__unused id<MSIDCacheItemSerializing>)serializer
-                                              context:(__unused id<MSIDRequestContext>)context
-                                                error:(__unused NSError **)error
+- (NSArray<MSIDCredentialCacheItem *> *)tokensWithKey:(MSIDCacheKey *)key
+                                           serializer:(id<MSIDCacheItemSerializing>)serializer
+                                              context:(id<MSIDRequestContext>)context
+                                                error:(NSError **)error
 {
-    NSArray *tokenItems = [self itemsWithKey:key context:context error:error];
-
-    if (!tokenItems)
+    NSArray<MSIDCredentialCacheItem *> *tokenList = [[NSArray alloc] init];
+    
+    if ([key isKindOfClass:([MSIDDefaultCredentialCacheKey class])])
     {
-        return nil;
+        /*
+         Lazy load app and shared credential blob to sync in memory cache with persistent cache.
+         */
+        [self initializeCredentialStorageItem:key serializer:serializer context:context error:error];
```

Comment:
as discussed offline, for RTs (shared storage items) we should always update keychain in case someone else updated it in the meantime and we have old cache. 
For ATs and IDs, the logic should be following:
1. Check in memory
2. If found, return from memory
3. If not found, read keychain
4. Try finding